modules/frontend/search_sharder.go

```diff
@@ -225,13 +225,54 @@ func (s *asyncSearchSharder) ingesterRequest(ctx context.Context, tenantID strin
 
 	// if ingester start == ingester end then we don't need to query it
 	if ingesterStart == ingesterEnd {
-		return nil, nil
+		return nil
 	}
 
 	searchReq.Start = ingesterStart
 	searchReq.End = ingesterEnd
 
-	return buildIngesterRequest(ctx, tenantID, parent, &searchReq)
+	// Split the start and end range into sub requests for each range.
```

Comment:
I was thinking if we can make this code simpler by making the min interval a fixed size??

I was thinking, something like this??

![image](https://github.com/grafana/tempo/assets/9503187/93d1b623-d28d-46ca-ac7c-460466bcdc10)

Reply:
https://github.com/grafana/tempo/pull/3574/commits/b8906ab6191d3a4d4310d267771d1b0e472762d4

I've pushed the above for this request.